Added tty support

sshbio opened the pull request:
I found this nice trick from [here](http://askubuntu.com/questions/147462/how-can-i-change-the-tty-colors) and [here](https://github.com/joepvd/tty-solarized).

And I use this a lot without root and without problem on Ubuntu, ArchLinux and Open-SUSE.

polyzen commented:
Shouldn't this make vim in a ptty match vim in a terminal emulator? Do I need to do anything else (aside from of course the proper vim setup)?

Had to change `elif [ "$TERM" != "linux" ]` to `elif [ "$TERM" != "linux" ]; then`, otherwise:

    ~ > source gruvbox_256palette.sh                
    gruvbox_256palette.sh:142: parse error near `fi'

sshbio commented:
Sorry for the `; then`

Well, this only replace the 8 colors available in the TTY (red, yellow, green, cyan, magenta, black, white) by those defined by the rgb code (along with the bold version).  It will not be possible to use as much colors as desired (only those 8, 16 by playing with bold properties), and vim will require configuration to use the appropriate color and avoid this [issue](http://stackoverflow.com/questions/19571274/vim-colors-in-tty-terminal).

With the exception of the lack of colors, the script should make vim totally similar to the original gruvbox theme.  Maybe you have a mac system: I will try the whole script in my TTY and tell if the colors properly shifted.

sshbio commented:
I checked the script, fixed it and adjusted the colors a few.

You can still adjust the colors (the bold colors are currently the dimmed one from your theme, and are not very visible but there is nothing brighter than normal colors in the theme).

Before script:
![img_20151004_132715](https://cloud.githubusercontent.com/assets/7250745/10267722/5a953c7a-6a8f-11e5-9a53-f9c570efd4b8.jpg)

After script:
![img_20151004_132802](https://cloud.githubusercontent.com/assets/7250745/10267724/6ba9c9d6-6a8f-11e5-9578-6dbda2436b4d.jpg)

morhetz commented:
Well, I'm not sure about this PR
Probably this could be added to the [gruvbox-contrib repo](https://github.com/morhetz/gruvbox-contrib)
But colors should be corrected to match the [16-colors pallete](https://github.com/morhetz/gruvbox-contrib/blob/master/color.table) and script should be splitted somehow to support both dark and light versions

sshbio commented:
It may be wiser.  Better give people bugless experience.

The tty terminal require to have different colors for bold text.  The one I chose for bold are on the right column.

Dark version
![terminal sexy_2015-11-04_17-27-34](https://cloud.githubusercontent.com/assets/7250745/10944401/43c9b050-8319-11e5-9150-836e777e2362.png)

Light version
![terminal sexy_2015-11-04_17-18-44](https://cloud.githubusercontent.com/assets/7250745/10944152/0479267a-8318-11e5-8167-aeb82f68db1a.png)

Is this better for you?  If it is, I can make a PR in gruvbox-contrib with two scrip[t for dark and light versions.

Here are the scripts produced by terminal.sexy

Dark version
```sh
#!/bin/sh
if [ "$TERM" = "linux" ]; then
  /bin/echo -e "
  \e]P0ebdbb2
  \e]P1cc241d
  \e]P298971a
  \e]P3d79921
  \e]P4458588
  \e]P5b16286
  \e]P6689d6a
  \e]P7282828
  \e]P8ebdbb2
  \e]P9fb4934
  \e]PAb8bb26
  \e]PBfabd2f
  \e]PC83a598
  \e]PDd3869b
  \e]PE8ec07c
  \e]PF3c3836
  "
  # get rid of artifacts
  clear
fi
```

Bright version
```sh
#!/bin/sh
if [ "$TERM" = "linux" ]; then
  /bin/echo -e "
  \e]P0fbf1c7
  \e]P1cc241d
  \e]P298971a
  \e]P3d79921
  \e]P4458588
  \e]P5b16286
  \e]P6689d6a
  \e]P73c3836
  \e]P8ebdbb2
  \e]P99d0006
  \e]PA79740e
  \e]PBb57614
  \e]PC076678
  \e]PD8f3f71
  \e]PE427b58
  \e]PF1d2021
  "
  # get rid of artifacts
  clear
fi
```

I can change them so that they look like the one you've written.

Thank you for your last answer!
